repository/zookeeper/0.1.0/framework.yaml

```
@@ -0,0 +1,39 @@
+name: "zookeeper"
+version: "0.1.0"
```

```suggestion
version: "0.2.0"
```

I think we need to follow SemVer 2 here and increment the Minor version as we introduce here the new structure from kep-9. It also suits under `0.2.0` the purpose for users using cli v0.2.0 can still install zk by setting the `--package-version=0.1.0` flag.

---

okay, I thought we're just breaking the compatibility and removing the old packages. If we don't want to do that and want to keep both... I am fine with that. Others have opinions here? @gerred 

If we go that way we probably should really communicate which version of package is for which kudo and not allow installing something that breaks.

Btw. we already made a backward incompatble change last time with switching to go templates. So current packages cannot be rendered with 0.1.0 kudo. I thought this is the same case...

---

https://semver.org/#spec-item-4

---

> okay, I thought we're just breaking the compatibility and removing the old packages. If we don't want to do that and want to keep both... I am fine with that. Others have opinions here? @gerred
> 
Well, at some point we need to start incrementing. And that point shouldn't be ideally 1.0.0 but earlier. I am still voting for trying to increment from now on each version SemVer 2 conform. See more on the reasons why we didn't the last time in my other answer below.

> If we go that way we probably should really communicate which version of package is for which kudo and not allow installing something that breaks.
> 
Exactly, imo that's why we have this line for: https://github.com/kudobuilder/frameworks/blob/3e02a65684d98470185ad32d33602a400db0cd2c/repository/zookeeper/0.1.0/framework.yaml#L3
To me, this line communicates with what KUDO version this Framework works.
I know that KUDO does ignore essentially this line right now. It would come now handy if it wouldn't ignore it :)

> Btw. we already made a backward incompatble change last time with switching to go templates. So current packages cannot be rendered with 0.1.0 kudo. I thought this is the same case...

Yes, and for me the main reason was, that we didn't have at that time a tool that could work with both. Now with https://github.com/kudobuilder/kudo/pull/343 merged for `0.3.0` the CLI supports for the first time both versions. Imo it would be a pity if we cannot show case that.
 
"If two people say the same thing, it's probably not the same thing".

---

@fabianbaier 

> Yes, and for me the main reason was, that we didn't have at that time a tool that could work with both. Now with kudobuilder/kudo#343 merged for 0.3.0 the CLI supports for the first time both versions. Imo it would be a pity if we cannot show case that.

Yeah that is not true :( we decided to drop the old format and introduce a breaking change. 0.3.0 can parse only 0.3.0 packages, not the old format

---

Agreed that it shouldn't be ignored. The CLI doesn't support both versions now btw, it only supports the new version.

Let's just all handshake agree this is the last time we make this change across all frameworks, and make it a high priority to implement semver matching for KUDO frameworks.

I want to, in general, cut out the repository itself as being versioned. Versions should be released out of master, so I'd even be ok bumping this up a level (just zookeeper, not zookeeper-0.1.0)